### ISSUE TITLE
requirePaddingNewLinesBeforeLineComments exception for switch statements

### ISSUE THREAD

**Issue by TheSavior** [jscs-dev/node-jscs#1312]
Switch statements currently require:

```javascript
switch(num) {
  case 4: 
    
    // comment
  default: 
    return false;
}
```

but should allow

```javascript
switch(num) {
  case 4: 
    // comment
  default: 
    return false;
}
```

We did the same thing with `firstAfterCurly` to allow blocks like this:

```javascript
if (true) {
  // comment
}
```

**Comment by markelog** [comment]
Why not? Is that something that could be used in one of the supported presets?

**Comment by TheSavior** [comment]
I'm sure Airbnb would want this one. It was the same case with `firstAfterCurly`.

**Comment by markelog** [comment]
Could post a link to relevant statement?

**Comment by TheSavior** [comment]
By relevant statement, do you mean confirmation that someone actually wants this rule value? I know we want this rule value at Wealthfront, @hshoff, any gut feeling as to how Airbnb feels?

**Comment by james-hart** [comment]
+1, I'd appreciate the implementation if it at possible.. perhaps "firstAfterColon"? :D

**Comment by markelog** [comment]
So does anyone would like to implement it?

**Comment by hzoo** [comment]
We released [3.0](https://github.com/jscs-dev/node-jscs/releases/tag/v3.0.0) recently! We also wrote a [post](https://medium.com/@markelog/jscs-end-of-the-line-bc9bf0b3fdb2#.oot3wvgpu) about how 3.0 will be our last major release (since we are deprecating the project) and that we are [merging with ESLint](http://eslint.org/blog/2016/04/welcoming-jscs-to-eslint)!

Thus we won't be working on any new features moving forward which includes new rules/options. We will continue to fix bugs and work on CST/better autofixing for current rules in the upcoming months both to help JSCS users and to gain experience that we can hopefully use in ESLint. But otherwise we both teams will be working on getting [ESLint feature parity with JSCS](https://github.com/eslint/eslint/milestones/JSCS%20Compatibility) and just a nice transition for our users.

I'm putting an `orphaned` label to note this issue isn't planned to be worked on, but might be a good contribution as a custom rule (not in JSCS core itself).

I would encourage anyone who wants the rule to try writing it themselves: I would either write a custom rule or submit an issue to ESLint and if accepted a PR. Otherwise a custom rule that won't be in core repo itself.

JSCS rules are also very similar to ESLint rules: here are some resources to get started: 

- [Project-specific Lint Rules with ESLint](https://gist.github.com/jareware/7179093)
- [Creating Custom Rules for ESLint](https://insideops.wordpress.com/2015/12/08/creating-custom-rules-for-eslint)
- [Maintaining JavaScript Code Quality with ESLint ](https://www.paypal-engineering.com/2014/12/12/maintaining-javascript-code-quality-with-eslint/)
